src/lib/support/BufferReader.h

```diff
@@ -58,77 +58,90 @@ class Reader
      */
     uint16_t OctetsRead() const { return static_cast<uint16_t>(mReadPtr - mBufStart); }
 
+    /**
+     * The reader status.  Once the status becomes a failure status, all later
+     * read operations become no-ops and the status continues to be a failure
+     * status.
+     */
+    CHIP_ERROR StatusCode() const { return mStatus; }
+
     /**
      * Read a single 8-bit unsigned integer.
      *
      * @param [out] dest Where the 8-bit integer goes.
      *
-     * @return Whether the read succeeded.  The read can fail if there are not
-     *         enough octets available.
+     * @note The read can put the reader in a failed-status state if there are
+     *       not enough octets available.  Callers must either continue to do
+     *       more reads on the return value or check its status to see whether
+     *       the sequence of reads that has been performed succeeded.
      */
     CHECK_RETURN_VALUE
```

Comment:
do these still need CHECK_RETURN_VALUE?

Should that be needed for StatusCode instead?

Reply:
These do still need `CHECK_RETURN_VALUE`, so people don't forget to call `StatusCode`.  Adding it to `StatusCode` is a good idea; I will do that.